src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java

```diff
@@ -443,10 +442,12 @@ public synchronized static void cancelBuild(WorkflowJob job, Build build) {
 
 	public synchronized static void cancelBuild(WorkflowJob job, Build build, boolean deleted) {
 		if (!cancelQueuedBuild(job, build)) {
-			cancelRunningBuild(job, build);
+      cancelRunningBuild(job, build);
 		}
-		if (deleted)
+		if (deleted) {
+			deleteRun(getRun(job, build));
```

Comment:
@gabemontero This is where the run gets deleted when a build is deleted.

Reply:
yep cool thanks ... see the other comment around relist/watch consistency and not relying soley on watch events while Jenkins is still up.

Reply:
i'm not clear on why this is here?  What's the scenario(what action did the user take)?

Reply:
(and if the build was deleted, why can't we rely on the other logic you added to delete the run?)

Reply:
I was just looking at that, will update.